### DISCUSSION TITLE
why  compute loss twice in Imagent example, isn't here overhead?

### DISCUSSION THREAD

**Discussion by wztdream** [google/flax#1569] [Q&A]
Hi,
Why re-compute loss in `compute_metrics` function in the imagenet example? Isn't here over head? Or does` jax.jit` mechanism can avoid the overhead?

defined here, line 85
https://github.com/google/flax/blob/d7172e9d3ea75a68a218ca68d07ee384d656664f/examples/imagenet/train.py#L84-L92

first compute loss here
https://github.com/google/flax/blob/d7172e9d3ea75a68a218ca68d07ee384d656664f/examples/imagenet/train.py#L121

re-compute here
https://github.com/google/flax/blob/d7172e9d3ea75a68a218ca68d07ee384d656664f/examples/imagenet/train.py#L146
